Add support for Xlayer

galaxy-ggatheca opened the pull request:
Main changes,
- Add dockerfile for op-node with debian base support, ie apt instead of apk
- Make the dockerfile used configurable with default options
- For op-geth init, add extra init args when needed, default empty
- For op-node binary allow path to be configurable instead of assuming default path; Default option available

galaxy-ggatheca committed:
Add debian version of op-node dockerfile

galaxy-ggatheca committed:
Support custom dockerfile for opnode and opgeth

galaxy-ggatheca committed:
Allow adding extra args for init geth, allow op-node bin to be configurable

galaxy-ggatheca committed:
add EL_INIT_EXTRAS to env variables opgeth

galaxy-ggatheca committed:
fix init geth

galaxy-ggatheca committed:
inti extra args position fix

galaxy-tbehrens reviewed (approved):
This looks good. Ofc XLayer is being special. Still better than their zkEVM Erigon stuff though, fingers crossed

galaxy-tbehrens commented on op-geth/docker-entrypoint.sh:
```diff
@@ -40,7 +40,7 @@ if [ -n "${GENESIS_URL}" ]; then
     fi
 
-    geth init --datadir=/var/lib/op-geth --state.scheme="${INIT_STATE_SCHEME}" genesis.json
+    geth ${EL_INIT_EXTRAS} init --datadir=/var/lib/op-geth --state.scheme="${INIT_STATE_SCHEME}" genesis.json
```
```suggestion
#shellcheck disable=SC2086
    geth ${EL_INIT_EXTRAS} init --datadir=/var/lib/op-geth --state.scheme="${INIT_STATE_SCHEME}" genesis.json
```

galaxy-tbehrens reviewed (changes requested):
Looks good overall. Lint is failing, suggest a shellcheck disable. We do not want double quotes in that location

galaxy-ggatheca committed:
Update op-geth/docker-entrypoint.sh

Lint is failing, suggest a shellcheck disable. We do not want double quotes in that location

Co-authored-by: galaxy-tbehrens <thorsten.behrens@galaxy.com>

galaxy-ggatheca committed:
ethd version use bin path for op-node

galaxy-ggatheca committed:
default value for OPNODE_BIN_PATH in ethd if not present
